File: solutions/2025/day01/part2.ts
export function solve(input: string): number | string {
  const lines = input.trim().split("\n");

  const rotations = lines.map((it) => toInt(it));

  let value = 50;
  let password = 0;

  for (const rotation of rotations) {
    for (const i of countTo(rotation)) {
      value = (100 + (value + i)) % 100;
      if (value === 0) {
        password += 1;
      }
    }
  }

  return password;
}

function toInt(line: string) {
  return parseInt(line.replace("L", "-").replace("R", "+"), 10);
}

function countTo(n: number) {
  const sign = n < 0 ? -1 : 1;
  return Array.from({ length: Math.abs(n) }, (_) => sign);
}

File: solutions/2025/day01/part2.py
def solve(input: str) -> int:
    pos, zeros = 50, 0
    for line in input.strip().split('\n'):
        dist = int(line[1:])
        if line[0] == 'L':
            zeros += dist // 100 if pos == 0 else (dist - pos) // 100 + 1 if dist >= pos else 0
            pos = (pos - dist) % 100
        else:
            zeros += dist // 100 if pos == 0 else (dist + pos - 100) // 100 + 1 if dist + pos >= 100 else 0
            pos = (pos + dist) % 100
    return zeros
